genie_registry/clinical.py

```diff
@@ -77,6 +77,92 @@ def _check_year(clinicaldf: pd.DataFrame, year_col: int, filename: str,
     return error
 
 
+def _check_int_dead_consistency(clinicaldf: pd.DataFrame) ->  str:
+    """Check if vital status interval and dead column are consistent
+
+    Args:
+        clinicaldf: Clinical Data Frame
+
+    Returns:
+        Error message if values and inconsistent or blank string
+    """
+    cols = ["INT_DOD", "DEAD"]
+    for col in cols:
+        # Return empty string is columns don't exist because this error
+        # is already handled.
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ''
+    is_dead = clinicaldf['DEAD'].astype(str) == "True"
+    is_alive = clinicaldf['DEAD'].astype(str) == "False"
+    allowed_str = ['Unknown', 'Not Collected', 'Not Applicable',
+                   "Not Released"]
+    is_str = clinicaldf['DEAD'].isin(allowed_str)
+    # Check that all string values are equal each other
+    is_equal = all(
+        clinicaldf.loc[is_str, "DEAD"] == clinicaldf.loc[is_str, "INT_DOD"]
+    )
+    # If dead, int column can't be Not Applicable
+    # If alive, int column can't have values
+    if (any(clinicaldf.loc[is_dead, 'INT_DOD'] == "Not Applicable") or
+            not all(clinicaldf.loc[is_alive, 'INT_DOD'].isin(allowed_str)) or
+            not is_equal):
+        return (
+            "Patient: you have inconsistent values in "
+            "INT_DOD, YEAR_DEATH, DEAD\n"
+        )
+    return ''
+
+
+def _check_int_year_consistency(
+        clinicaldf: pd.DataFrame,
+        cols: list, string_vals: list
+    ) ->  str:
+    """
+    Check if vital status interval and year columns are consistent in
+    their values
+
+    Args:
+        clinicaldf: Clinical Data Frame
+        cols: Columns in the clinical data frame
+        string_vals: String values that aren't integers
+
+    Returns:
+        Error message if values and inconsistent or blank string
+    """
+    interval_col = ''
+    year_col = ''
+    for col in cols:
+        # This is assuming that interval and year columns start with
+        # INT/YEAR
+        interval_col = col if col.startswith("INT") else interval_col
+        year_col = col if col.startswith("YEAR") else year_col
+        # Return empty string is columns don't exist because this error
+        # is already handled.
+        if not process_functions.checkColExist(clinicaldf, col):
+            return ""
+
+    is_inconsistent = False
```

Comment:
to separate out the two types of inconsistencies, maybe define two flags here:

1. is_missing_inconsistent
2. is_redaction_inconsistent

and set flags in appropriate locations.